scripts/init.sql

```diff
@@ -1,289 +1,228 @@
-
-ALTER TABLE wishcards
-ADD
-    CONSTRAINT wishcards_created_by_fkey FOREIGN KEY (created_by) REFERENCES users (id);
-
-ALTER TABLE wishcards
-ADD
-    CONSTRAINT wishcards_item_id_fkey FOREIGN KEY (item_id) REFERENCES items (id);
-
-ALTER TABLE messages
-ADD
-    CONSTRAINT messages_wishcard_id_fkey FOREIGN KEY (wishcard_id) REFERENCES wishcards (id);
-
-ALTER TABLE wishcards
-ADD
-    CONSTRAINT wishcards_image_id_fkey FOREIGN KEY (image_id) REFERENCES images (id);
-
-ALTER TABLE children
-ADD
-    CONSTRAINT children_image_id_fkey FOREIGN KEY (image_id) REFERENCES images (id);
-
-ALTER TABLE items
-ADD
-    CONSTRAINT items_image_id_fkey FOREIGN KEY (image_id) REFERENCES images (id);
-
-ALTER TABLE agencies
-ADD
-    CONSTRAINT agencies_image_id_fkey FOREIGN KEY (image_id) REFERENCES images (id);
-
-ALTER TABLE community_posts
-ADD
-    CONSTRAINT community_posts_image_id_fkey FOREIGN KEY (image_id) REFERENCES images (id);
-
-ALTER TABLE users
-ADD
-    CONSTRAINT users_image_id_fkey FOREIGN KEY (image_id) REFERENCES images (id);
-
-ALTER TABLE orders
-ADD
-    CONSTRAINT orders_donor_id_fkey FOREIGN KEY (donor_id) REFERENCES users (id);
-
-ALTER TABLE orders
-ADD
-    CONSTRAINT orders_child_id_fkey FOREIGN KEY (child_id) REFERENCES children (id);
-
-ALTER TABLE orders
-ADD
-    CONSTRAINT orders_item_id_fkey FOREIGN KEY (item_id) REFERENCES items (id);
-
-ALTER TABLE orders
-ADD
-    CONSTRAINT orders_agency_id_fkey FOREIGN KEY (agency_id) REFERENCES agencies (id);
-
-ALTER TABLE wishcards
-ADD
-    CONSTRAINT wishcards_order_id_fkey FOREIGN KEY (order_id) REFERENCES orders (id);
-
--- functions
-CREATE OR REPLACE FUNCTION public.TRIGGER_SET_UPDATED_DATE()
-    RETURNS TRIGGER
-    LANGUAGE PLPGSQL
+CREATE TABLE "agencies" (
+    "id" uuid PRIMARY KEY NOT NULL DEFAULT (gen_random_uuid()),
+    "name" varchar(255) NOT NULL,
+    "website" varchar(255) NOT NULL,
+    "phone" varchar(100) NOT NULL,
+    "bio" varchar(500),
+    "address_line_1" varchar(255) NOT NULL,
+    "address_line_2" varchar(255),
+    "city" varchar(100) NOT NULL,
+    "state" varchar(100) NOT NULL,
+    "country" varchar(100) NOT NULL,
+    "zip_code" varchar(50) NOT NULL,
+    "verified" boolean NOT NULL DEFAULT false,
+    "employer_identification_number" varchar(255) UNIQUE,
+    "account_manager_id" uuid NOT NULL,
+    "image_id" uuid NOT NULL,
+    "created_at" timestamp NOT NULL DEFAULT (CURRENT_TIMESTAMP),
+    "updated_at" timestamp
+);
+
+CREATE TABLE "children" (
+    "id" uuid PRIMARY KEY NOT NULL DEFAULT (gen_random_uuid()),
+    "first_name" varchar(255) NOT NULL,
+    "last_name" varchar(255) NOT NULL,
+    "birth_year" integer NOT NULL,
+    "interest" varchar(255) NOT NULL,
+    "story" varchar(255) NOT NULL,
+    "image_id" uuid NOT NULL,
+    "created_at" timestamp NOT NULL DEFAULT (CURRENT_TIMESTAMP)
+);
+
+CREATE TABLE "community_posts" (
+    "id" uuid PRIMARY KEY NOT NULL DEFAULT (gen_random_uuid()),
+    "message" varchar(255) NOT NULL,
```

Comment:
@stacysealky `VARCHAR(255)` doesn't mean it stores 255 characters, it means it's the max number of characters that can be counted by an 8-bit number, plus an additional byte as an offset.

[This StackOverflow post](https://stackoverflow.com/questions/1217466/is-there-a-good-reason-i-see-varchar255-used-so-often-as-opposed-to-another-l) describes it in a bit more detail, but tl;dr, if you don't know how much data you're going to be storing in a column but don't want to set a hard restriction, you use `VARCHAR(255)`.

Reply:
Are you sure? Wont the string get truncated if it has over 255 characters? 

I would suggest change it to a text type or not specifying any length at all (so just varchar :) )

Reply:
@Tokikko Actually, I think you're right, I may have gotten some wires crossed comparing it to SQL Server.

@Enubia I just checked some of my notes and recall speaking about this to a DBA and determining that `text` is a better type for any column storing a string of an undetermined length, as mentioned by @Tokikko.

Would you be open to swapping everything to `text` instead?

Reply:
Sure thing

Reply:
wait wait, but we do want to have a max limit for all string inputs in db, right?
I mean there's a frontend validation for each string input. For example, in the form, it allows 255 chars max for first name, 2000 for the child story, etc. 
So shouldn't we want to keep the rules consistent?

Reply:
@stacysealky the parameter used for `VARCHAR` is for bytes, not characters, so we would need to add some arbitrary buffer on top of whatever we set the limit to. It would also likely be quite bothersome to change it later on if we changed the validation rules.

We will implement length validation rules on the backend to prevent entries that exceed the max limit as a secondary safeguard.

Reply:
Ok sounds good. I'm in favor of changing all to TEXT.

Reply:
@nick-w-nick 

varchar and text do not really differ in its type

> The PostgreSQL Text data type is used to keep the character of infinite length. And it can hold a string with a maximum length of 65,535 bytes.
> It is the same as the PostgreSQL Varchar data type, rather without any argument for mentioning the length of the character ie., the PostgreSQL Varchar Data Type without n is the same as the Text data type.